### ISSUE TITLE
STM32 I2C driver "repeated start" issue

### ISSUE THREAD

**Issue by jmchiappa** [stm32duino/Arduino_Core_STM32#583]
Hello,

I'm trying to connect to the MMA8451 accelerometer. This chip requires an I2C driver with repeated-start feature enabled to discuss with.
I've  a NUCLEO-F411RE, lib 1.6.1 and no repeated start is generated due to some low level limitations.
Let me explain: 
XferOptions variable is always set to `I2C_NO_OPTION_FRAME` in the `HAL_I2C_Transmit_IT`. So, once the Xfer is completed, it fires `I2C_MasterTransmit_TXE`. As the following condition is always false (stm32f4xx_hal_i2c.c):
`if ((CurrentXferOptions != I2C_FIRST_AND_LAST_FRAME) && (CurrentXferOptions != I2C_LAST_FRAME) && (CurrentXferOptions != I2C_NO_OPTION_FRAME))`
a STOP will be generated (line 5025):
```
      /* Disable EVT, BUF and ERR interrupt */
      __HAL_I2C_DISABLE_IT(hi2c, I2C_IT_EVT | I2C_IT_BUF | I2C_IT_ERR);

      /* Generate Stop */
      SET_BIT(hi2c->Instance->CR1, I2C_CR1_STOP);

```
Does someone can confirm ?
How could we override XferOptions by setting to `I2C_LAST_FRAME_NO_STOP` ?
Ok, I understand that it's not directly related to the framework by itself, but I observed that I2C driver is really different regarding the variant.
In other word, by using `TwoWire::endTransmission(false)`, it should not send a STOP condition (instead of UNUSED(endStop) ).

Thank you,
jean-marc

**Comment by fpistm** [comment]
Hi Jean-Marc,
thanks for the report. Will try to investigate when got a time slot. (will be on vacation soon)
Maybe @LMESTM will have a look when he came back from vacation.

**Comment by jmchiappa** [comment]
Hi Frédéric, 
Thank you for your reply.
After deeper investigation, it seems that `HAL_I2C_Master_Seq_Transmit_IT` is more relevant to generate a 'repeated-start' as XferOptions is passed in argument list.
Curiously, this function is called `HAL_I2C_Master_Sequential_Transmit_IT` in F0,F1,F2,F3 series but it's correctly bind in legacy file. 
I'll have a try as soon as I can (in holidays soon too).

Enjoy your holidays !

**Comment by fpistm** [comment]
Thanks, enjoy too.

The function has been renamed in newer STM32CubeFW Release for or MISRA-C 2012 compliancy:
See the [Release_Notes.html](https://github.com/stm32duino/Arduino_Core_STM32/blob/master/system/Drivers/STM32F4xx_HAL_Driver/Release_Notes.html)

I guess at term all series will be renamed.  The next cube update add this too for some of this series, see #566.

**Comment by fpistm** [comment]
Fixed by #590

**Comment by fpistm** [comment]
During non reg tests for 1.7.0 release found issue with STM32 F2 and F4 related to #590. 
So I've reverted it to perform further investigation.
